spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
>The PayloadApplicationEvent is something to which the raw object is wrapped when it is published.

Yes, I understand that, but see my comment from the javadoc of `PayloadApplicationEvent`.

I don't think we expect users to publish arbitrary object as events.

I still feel this adapter should emit nothing but events and not unwrap.